cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ListenersValidator.java

```diff
@@ -656,6 +661,31 @@ private static void validateOauth(Set<String> errors, GenericKafkaListener liste
         }
     }
 
+    private static void validateAndOverrideServiceAccountOAuth(KafkaListenerAuthenticationOAuth oAuth) {
+        if (oAuth.getValidIssuerUri() == null) {
+            oAuth.setValidIssuerUri("https://kubernetes.default.svc.cluster.local");
+        }
+        if (oAuth.getJwksEndpointUri() == null) {
+            oAuth.setJwksEndpointUri("https://kubernetes.default.svc.cluster.local/openid/v1/jwks");
+        }
+        if (oAuth.getServerBearerTokenLocation() == null) {
+            oAuth.setServerBearerTokenLocation("/var/run/secrets/kubernetes.io/serviceaccount/token");
+        }
+
+        if (oAuth.getIncludeAcceptHeader() == null || oAuth.getIncludeAcceptHeader()) {
+            if (oAuth.getIncludeAcceptHeader() != null && oAuth.getIncludeAcceptHeader()) {
+                LOGGER.warnOp("'includeAcceptHeader' force-set to 'false' for compatibility with 'serviceaccount-oauth'");
+            }
+            oAuth.setIncludeAcceptHeader(false);
+        }
+        if (oAuth.getCheckAccessTokenType() == null || oAuth.getCheckAccessTokenType()) {
+            if (oAuth.getCheckAccessTokenType() != null && oAuth.getCheckAccessTokenType()) {
+                LOGGER.warnOp("'checkAccessTokenType' force-set to 'false' for compatibility with 'serviceaccount-oauth'");
+            }
+            oAuth.setCheckAccessTokenType(false);
+        }
+    }
+
```

Comment:
I had second thoughts about that but I couldn't find a better way to perform the initialisation of these fields after CR configured values have already been applied. Validation is the logic to always be called at least once after all the values have been set and invoking this method multiple times doesn't change the outcome. This approach is least disruptive and gets the work done (unless I'm missing something very important there that's not completely obvious).

I'm not sure about the proposed alternatives. I guess the `validIssuerUri`, `jwksEndpointUri`, `serverBearerTokenLocation` can be set by overriding setter and getter methods with new default value annotations. Options `includeAcceptHeader` and `checkAccessTokenType` have to be fixed to non-default values, maybe that can be achieved by hardcoding a value into a getter override? But then there is this issue of silently ignoring the attempt to override these values - that's why the WARN logging is logged if an explicit attempt to set the wrong value of these attributes is detected.

Considering the Kafka configuration builder approach - I guess the challenge would again be with `includeAcceptHeader` and `checkAccessTokenType` as we have to make it impossible for the user to override the configuration with a new value - i.e. we have to perform the check and override after all the attributes have been set. These values should be set to defaults and not overridden with a different value.

Reply:
By having `KafkaClientAuthenticationServiceAccountOAuth` extend `KafkaClientAuthenticationOAuth`, and `KafkaListenerAuthenticationServiceAccountOAuth` extend `KafkaListenerAuthenticationOAuth` the same code that works to validate 'oauth' can be used to validate 'serviceaccount-oauth'. Otherwise that code would essentially have to be duplicated.

Reply:
After thinking some more about this I made an alternative implementation which does away with API duplication.
See: https://github.com/strimzi/strimzi-kafka-operator/pull/10260